test/uk/gov/hmrc/customsservicestatus/services/AdminCustomsStatusServiceSpec.scala

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.customsservicestatus.services
+
+import com.mongodb.client.result.InsertOneResult
+import org.bson.BsonValue
+import org.mockito.ArgumentMatchers.any
+import org.mockito.Mockito.when
+import uk.gov.hmrc.customsservicestatus.errorhandlers.AdminCustomsServiceStatusError
+import uk.gov.hmrc.customsservicestatus.errorhandlers.AdminCustomsServiceStatusError.*
+import uk.gov.hmrc.customsservicestatus.helpers.BaseSpec
+import uk.gov.hmrc.customsservicestatus.models.DetailType.*
+import uk.gov.hmrc.customsservicestatus.models.UnplannedOutageData
+
+import java.time.Instant
+import scala.concurrent.Future
+
+class AdminCustomsStatusServiceSpec extends BaseSpec {
+  trait Setup {
+    val service = new AdminCustomsStatusService(mockConfig, mockAdminCustomsServiceStatusRepository)
+    val validOutageUnplannedData: UnplannedOutageData = UnplannedOutageData(
+      InternalReference("Test reference"),
+      Preview("Test preview"),
+      Instant.now(),
+      None
+    )
+    case class acknowledgedInsertOneResult(isAcknowledged: Boolean = true) extends InsertOneResult {
+      override def wasAcknowledged(): Boolean   = isAcknowledged
+      override def getInsertedId:     BsonValue = null
+    }
+  }
+
+  "adminCustomsStatusService" should {
```

Comment:
I would use the name of the method under test here at line 47 rather than the name of the class.